pytest_fixtures/component/provision_azure.py

```diff
@@ -9,6 +9,45 @@
 from robottelo.constants import AZURERM_RHEL7_FT_GALLERY_IMG_URN
 from robottelo.constants import AZURERM_RHEL7_FT_IMG_URN
 from robottelo.constants import AZURERM_RHEL7_UD_IMG_URN
+from robottelo.constants import DEFAULT_ARCHITECTURE
+
+
+@pytest.fixture(scope='session')
+def sat_azure(request, session_puppet_enabled_sat, session_target_sat):
+    hosts = {'sat': session_target_sat, 'puppet_sat': session_puppet_enabled_sat}
+    yield hosts[request.param]
+
+
+@pytest.fixture(scope='module')
+def sat_azure_org(sat_azure):
+    yield sat_azure.api.Organization().create()
+
+
+@pytest.fixture(scope='module')
+def sat_azure_loc(sat_azure):
+    yield sat_azure.api.Location().create()
+
+
+@pytest.fixture(scope='module')
+def sat_azure_domain(sat_azure, sat_azure_loc, sat_azure_org):
+    yield sat_azure.api.Domain(location=[sat_azure_loc], organization=[sat_azure_org]).create()
+
+
+@pytest.fixture(scope='module')
+def sat_azure_default_os(sat_azure):
+    """Default OS on the puppet-enabled Satellite"""
+    search_string = 'name="RedHat" AND (major="6" OR major="7" OR major="8")'
+    return sat_azure.api.OperatingSystem().search(query={'search': search_string})[0].read()
+
+
+@pytest.fixture(scope='module')
+def sat_azure_default_architecture(sat_azure):
+    arch = (
+        sat_azure.api.Architecture()
+        .search(query={'search': f'name="{DEFAULT_ARCHITECTURE}"'})[0]
+        .read()
+    )
```

Comment:
I feel it would be good to include such default entities under satellite class or in satellite_mixins rather than having this into fixtures, similar to the `internal_capsule` property, non-blocking atm!

@jyejare @JacobCallahan ^

Reply:
Lets not overcrowd mixins with default entity helpers.